src/test/ui/rfc-2632-const-trait-impl/const-drop-bound.rs

```diff
@@ -0,0 +1,20 @@
+// check-pass
+
+#![feature(const_trait_impl)]
+#![feature(const_fn_trait_bound)]
+#![feature(const_precise_live_drops)]
+
+const fn foo<T, E>(res: Result<T, E>) -> Option<T> where E: ~const Drop {
+    match res {
+        Ok(t) => Some(t),
+        Err(_e) => None,
+    }
+}
+
+pub struct Foo<T>(T);
+
+const fn baz<T: ~const Drop, E: ~const Drop>(res: Result<Foo<T>, Foo<E>>) -> Option<Foo<T>> {
```

Comment:
So `~const Drop` bounds already are unlike `Drop` bounds in that the type does not have to actually implement the `Drop` trait. Is this noted anywhere?

Reply:
See https://internals.rust-lang.org/t/pre-rfc-revamped-const-trait-impl-aka-rfc-2632/15192#drop-5

Reply:
Ok. That makes sense now, seeing as there isn't really any other way to easily do it.